wgpu-hal/src/dx12/mod.rs

```diff
@@ -238,6 +238,9 @@ struct DeviceShared {
     heap_samplers: descriptor::GeneralHeap,
 }
 
+unsafe impl Send for DeviceShared {}
+unsafe impl Sync for DeviceShared {}
+
```

Comment:
question: Wait, why was this necessary? 👀 I think it would be good to have commentary explaining why.

Reply:
This also applies to similar changes in `wgpu-hal/src/gles/wgl.rs`.

Reply:
The suppressions in `wgpu` are because we write cross platform code and types on wasm aren't send/sync, so clippy gets mad at us for putting a !Send thing in a Arc.

These are different - we had a type, like `Device` which was unsafely Send/Sync'd, which contained an Arc<DeviceShared>. This meant we could use it cross thread, but as far the arc is concerned, its contents are !Send/Sync. This just makes the contents Send/Sync. It doesn't change anything, was just a previous oversight.